Run mypy CI with Python 3.8 too

We've noticed that `from __future__ import annotations` is not a panacea. For example:

```python
from __future__ import annotations
import numpy as np

CoolDType = type[np.int32] | type[np.float32]
```

mypy is happy with this in Python 3.11 but in Python 3.8:

```sh
test.py:4: error: Invalid type alias: expression is not a valid type  [valid-type]
test.py:4: error: The type "Type[type]" is not generic and not indexable  [misc]
Found 2 errors in 1 file (checked 1 source file)
```

To work this has to be `CoolDType = Union[type[np.int32], type[np.float32]]` in Python 3.8. There are probably as yet undiscovered instances of this disparity and the errors aren't very indicative of the issue so CI should catch them.

--- a/.github/workflows/mypy.yml
+++ b/.github/workflows/mypy.yml
@@ -14,6 +14,9 @@ on:
 
 jobs:
   mypy:
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.11"]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
@@ -23,10 +26,10 @@ jobs:
       - name: Setup xtgeo
         uses: "./.github/actions/setup_xtgeo"
         with:
-          python-version: 3.11
+          python-version: ${{ matrix.python-version }}
 
       - name: List dependencies
         run: pip freeze
 
-      - name: run mypy
+      - name: Run mypy
         run: mypy .

